src/main/java/ch/njol/skript/expressions/ExprLocationVectorOffset.java

```diff
@@ -70,8 +73,32 @@ public Class<? extends Location> getReturnType() {
 	}
 
 	@Override
-	public String toString(@Nullable Event e, boolean debug) {
-		return location.toString() + " offset by " + vectors.toString();
+	public String toString(@Nullable Event event, boolean debug) {
+		return location.toString(event, debug) + " offset by " + vectors.toString(event, debug) + (usingLocalAxes ? " using local axes" : "");
+	}
+
+	/**
+	 * Returns a location offset from the given location, adjusted for the location's rotation.
+	 * <p>
+	 * This behaves similarly to Minecraft's {@code /summon zombie ^ ^ ^1} command,
+	 * where the offset is applied relative to the entity's facing direction.
+	 *
+	 * @see <a href="https://minecraft.wiki/w/Coordinates#Local_coordinates">Local Coordinates</a>.
+	 * @param loc The location
+	 * @param offset The offset
+	 * @return The offset location
+	 */
+	private static Location getFacingRelativeOffset(Location loc, Vector offset) {
+		Vector forward = loc.getDirection();
+		Vector up = new Vector(0, 1, 0);
```

Comment:
this isn't actually the local up direction

Reply:
You may want to replace this whole method with Quaternionf.rotationZYX and use that to transform the vector